### ISSUE TITLE
Enabling rule MA0148 creates a conflict with the Regex Source Generator

### ISSUE THREAD

**Issue by evgeshin** [meziantou/Meziantou.Analyzer#747]
### Version of the Meziantou.Analyzer NuGet package

2.0.163

### Rule Identifier

MA0148

### Target Framework

net8.0

### C# Language version

C# 12

### Description

For several rules (including MA0148) enabled generation of diagnostics for generated files:

``context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);``

This makes it difficult to use these rules due to conflicts in the generated code

### Reproduction Steps

Minimal code:

Add next lines to project file:

```
    <PropertyGroup>
        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
    </PropertyGroup>

    <ItemGroup>
        <PackageReference Include="Meziantou.Analyzer" Version="2.0.163"/>
    </ItemGroup>

    <ItemGroup>
        <GlobalAnalyzerConfigFiles Include="my.globalconfig"/>
    </ItemGroup>
```

Add my.globalconfig with next content:
```
is_global = true

dotnet_diagnostic.MA0110.severity = warning
dotnet_diagnostic.MA0148.severity = warning
```

Add new class with next content:
```c#
using System.Text.RegularExpressions;

namespace ConsoleApp2;

internal static partial class Test
{
    [GeneratedRegex("Part[.]of[.].+[.]string[.]", RegexOptions.None, 1000)]
    private static partial Regex TextRegex();
}
```

Try build project

**Comment by meziantou** [comment]
Fixed by https://github.com/meziantou/Meziantou.Analyzer/commit/3a304f44fc222a42a2062d607e37c8dbfc6d8b4f
